### PR TITLE
Updates the output tables font colours

### LINKED ISSUE #1062: Colouration unclear in output tables
**Is your feature request related to a problem? Please describe.**
The text over the default colourmaps for the output tables are unclear with the white font. 

**Describe the solution you'd like**
There are several possible approaches to remedy this:
- Change the colour-maps to be less vibrantly coloured by default (e.g. Blue to Red rather than white)
- Change the font dynamically based on the underlying colour-map value (e.g. if colourmap.colour > (128, 128, 128): text.colour = (0,0,0)

Possibly others

**Describe alternatives you've considered**
Discussed above

**Additional context**
[Colour maps](https://fitbenchmarking.readthedocs.io/en/stable/users/options/plotting_option.html) found here mostly demonstrate this at the "good" end with the white text.

### PR DESCRIPTION
**Pull request by RabiyaF**
Fixes #1062 

**Description of Work**

The text colour in the 4 output tables is selected based on the contrast ratio. The contrast ratio is calculated according to the Web Content Accessibility Guidelines (WCAGs).

[Contrast ratio](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#dfn-contrast-ratio)
[Relative Luminosity](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#dfn-relative-luminance)

The contrast ratio is calculated between the background colourmap and the foreground text option of either white or black text. White is the default/preferred option. 

It is selected if it meets the AAA WCAGs for contrast ratio. If the requirement is not met, the text colour with the greatest contrast is selected between the two available options.  

**Function: Does the change do what it's supposed to?**

The output tables are now displayed with more accessible text which should be readable even when non-default colormaps are used by users.

**Tests: Does it pass? Is there adequate coverage for new code?**

The expected results (acc.html, compare.html, runtime.html and local_min.html) for test_tables.py have been updated.
New tests are also added for the contrast ratio calculations.

### CODE REVIEW

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
This looks really good. I tested it locally and it looks to do the right thing, did you get any feedback from Jacob on this?

I only had some minor points.

File: fitbenchmarking/results_processing/base_table.py [hunk 1]
```diff
@@ -238,7 +245,21 @@ def get_str_result(self, result, html=False):
...
+            if isinstance(text_col, tuple):
+                ftext, stext = text_col
+                val_str = val_str.split('<br>')
+                val_str = (f'<a {text_class[ftext]} '
+                           f'href="{self.get_link_str(result)}">'
+                           f'{val_str[0]}</a>'
+                           f'<a {text_class[stext]} '
+                           f'href="{self.get_link_str(result)}">'
+                           f'{val_str[1]}</a>')
+            else:
```
[the remark this answers is not in the document]
Comment (RabiyaF) [reply fitbenchmarking/results_processing/base_table.py:259]: I defined the `get_hyperlink()` method in the base class and then overwrote it in `compare_table.py`. It uses the following snippet:

`def get_hyperlink(self, result, val_str, text_col):`
`         val_str = val_str.split('<br>')`
`         val_str = [super().get_hyperlink(result, val, col)
                   for val, col in zip(val_str, text_col)]`
`         val_str = ''.join(val_str)`
`         return val_str`

It works. However, the tests fail. For now I will use a `get_hyperlink()` definition in `compare_table.py` that does not call the super method.

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
This looks great! Looks like there're a few conflicts because the tests for outputs have been changed slightly. Would you mind regenerating the tables then I think we can merge this. 🚀

**Comment by RabiyaF** [comment]
I think that I have managed to resolve the merge conflicts. But I am not sure why the checkpointing test is failing on Windows.

**Review by AndrewLister-STFC — APPROVED** [review]
Looks great! I've opened a PR to fix the windows tests issue (which is nothing to do with this PR).

**Pull request merged**